src/content/blog/ddev-name-resolution-wildcards.md

```diff
@@ -0,0 +1,95 @@
+
+When your browser gets a URL, no matter how it gets it, and you want to display the page at that URL, the browser has to parse the URL, then turn the hostname into an IP address so it can find its way to it on the internet, then do the proper HTTP request to the internet site, including the URI. The key thing for us right now is that it has to look up that hostname.
+
+Normally with a DDEV local development project, we'll see URLs like `https://something.ddev.site`. It's all done like that so that your local project will work with real URLs and real HTTPS, and essentially behave like a real site. In the past, local development environments often used URLs like `http://127.0.0.1:2302/somesite, which isn't easy to look at and definitely doesn't behave the same in the browser as a "real" URL.
+
+## How DDEV hostnames get resolved
+
+But DDEV URLs have to have hostnames in them that a browser can resolve. How do we get that? How can we get a URL like `https://something.ddev.site`?
+
+When a browser tries to resolve `https://something.ddev.site` it looks up the hostname `something.ddev.site`. Normally that's a lookup in the DNS (Domain Name System) world. You don't have to know about DNS to understand that. The browser asks the internet "What does `something.ddev.site` resolve to, and the internet DNS system says "Oh, anything with `ddev.site` has the IP address `127.0.0.1`". So the browser nows to connect to your local machine (which is what `127.0.0.1` or `localhost` always means). This process is invisible to you, and works because the DDEV project maintains the DNS records for `ddev.site` on the internet. You don't have to know it and you don't have to think about it.
+
+## What is `ddev.site` and where does it come from?
+
+`ddev.site` and everything under it, like `you.ddev.site` and `something.else.entirely.ddev.site` are all (when you have working internet) records in the Domain Name System, maintained by the DDEV project. You don't have to think about them. You can `ping -c 1 something.ddev.site` and it will show that it is pinging `127.0.0.1`
+
+## What happens when you don't have internet?
+
+When you don't have internet capability, or DNS name resolution is broken, then DDEV's fallback during `ddev start` is to try to add the hostname to your `hosts` file. This is `/etc/hosts` on a Linux or macOS machine, or `C:\Windows\system32\drivers\etc\hosts` on a Windows machine. This is one of the very few times that DDEV will ever try to change the configuration of your local workstation. In general, the philosophy is never to do that. DDEV knows how to edit the hosts file properly and does it when you give it permission with your `sudo` password.
```

Comment:
```suggestion
In case there is no internet connection available, or the DNS name resolution is broken, as a fallback DDEV is trying to add the hostname to your `hosts` file during the startup (`ddev start`). This is `/etc/hosts` on a Linux or macOS based machine and `C:\Windows\system32\drivers\etc\hosts` on a Windows based one. This is one of the very few times that DDEV will ever try to change the configuration of your local workstation. In general, the philosophy is never to do that. DDEV knows how to edit the hosts file properly and does it when you give it permission with your `sudo` password.
```

Reply:
```suggestion
In case there is no internet connection available or the DNS name resolution is broken, as a fallback DDEV tries to add the hostname to your `hosts` file during the startup (`ddev start`). This is `/etc/hosts` in Linux/Mac OS X and `C:\Windows\system32\drivers\etc\hosts` in Windows. This is one of the very few times that DDEV will ever try to change the configuration of your local workstation. In general, the philosophy is never to do that. DDEV knows how to edit the hosts file properly and does it when you give it permission with your `sudo` password.
```

Reply:
```suggestion
When no internet connection available, or the DNS name resolution is broken, as a fallback DDEV tries to add the hostname to your `hosts` file during startup (`ddev start`). This is `/etc/hosts` on a Linux or macOS based machine and `C:\Windows\system32\drivers\etc\hosts` on a Windows based one. This is one of the very few times that DDEV will ever try to change the configuration of your local workstation. In general, the philosophy is never to do that. DDEV knows how to edit the hosts file properly and does it when you give it permission with your `sudo` password.
```